solution.rb

```diff
@@ -0,0 +1,29 @@
+TEXT = File.read('speech.txt')
+
+def find_words text
+  words_regex = /\w+[a-z']*\w+/
+  text.downcase.scan words_regex
+end
+
+def count_words words_array
+  Hash.new(0).tap do |word_count|
+    words_array.each { |word| word_count[word] += 1 }
+  end
+end
+
+def sort_words words_frequency_hash
+  words_frequency_hash.sort_by { |word, count| -count }
```

Comment:
:+1: Don't think I've seen anyone implement it this way before. I was putz'ing around with irb and `{}.sort_by(&:last).reverse` will do the same too.